.test/provenance/out.json

```
@@ -107,7 +107,7 @@
 			},
 			"runDetails": {
 				"builder": {
-					"id": "https://github.com/docker-library/meta/.github/workflows/build.yml@refs/heads/subset"
+					"id": "https://github.com/docker-library/meta/.github/workflows/build.yml@refs/heads/main"
```

This might seem like a wild idea but thoughts on versioning by branch name? It would solve the problem of needing to update `builder.id` if we make a change that warrants a new `builder.id` by incrementing the branch name.

```suggestion
					"id": "https://github.com/docker-library/meta/.github/workflows/build.yml@refs/heads/v1"
```

---

Would all changes that could affect the `builder.id` be only directly on `meta` though? Could changes here in `meta-scripts` also warrant a `builder.id` change?

---

Yes, changes in `meta-scripts` could invalidate the SLSA build model that would result in the need to rotate keys and change the `builder.id`. The reason to assign the version from `meta` is that `meta` includes `meta-scripts` as a git submodule and defines the top-level workflow file which is considered the "trusted control plane". 

It seems like a good place to include versioning and we've talked a bit about it in before. It isn't completely necessary but an idea.

---

It's a great thought, thanks for raising it!

Given changes here get automatically consumed into `meta` via the automation, I think I'd rather switch to maintaining a new `builder.id` value here in the code when we need to, because that's a lot easier to coordinate correctly than swapping the branch / branch name.

---

Sound good 😊